src/id/samehadaku/src/eu/kanade/tachiyomi/animeextension/id/samehadaku/Samehadaku.kt

```diff
@@ -177,22 +163,44 @@ class Samehadaku : ConfigurableAnimeSource, AnimeHttpSource() {
             add("nume", element.attr("data-nume"))
             add("type", element.attr("data-type"))
         }.build()
+
         return client.newCall(POST("$url/wp-admin/admin-ajax.php", body = form))
             .execute()
             .let {
-                val link = it.body.string().substringAfter("src=\"").substringBefore("\"")
-                val server = element.selectFirst("span")!!.text()
-                Pair(server, link)
+                val body = it.body.string()
+                val quality = element.selectFirst("span")!!.text()
+                val link = body.substringAfter("src=\"").substringBefore("\"")
+                Pair(quality, link)
             }
     }
 
-    private fun getVideosFromEmbed(server: String, link: String): List<Video> {
+    private fun getVideosFromEmbed(quality: String, link: String): List<Video> {
         return when {
             "wibufile" in link -> {
+                val videoQuality = when {
+                    "480" in quality -> "Wibufile 480p"
+                    "720" in quality -> "Wibufile 720p"
+                    "1080" in quality -> "Wibufile 1080p"
+                    else -> "Unknown Resolution"
+                }
+
+                if (".mp4" in link.lowercase()) {
+                    return listOf(Video(link, videoQuality, link, headers))
+                }
+
                 client.newCall(GET(link)).execute().use {
                     if (!it.isSuccessful) return emptyList()
-                    val videoUrl = it.body.string().substringAfter("cast(\"").substringBefore("\"")
-                    listOf(Video(videoUrl, server, videoUrl, headers))
+
+                    val body = it.body.string()
+                    val json = JSONObject(body.substringAfter("source = ").substringBefore(";"))
```

Comment:
Use [kotlinx.serialization](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/basic-serialization.md) and serializable classes instead of `JSONObject`